cmake/Modules/FindMKLDNN.cmake

```diff
@@ -118,6 +118,12 @@ ENDIF()
 SET(WITH_TEST FALSE CACHE BOOL "" FORCE)
 SET(WITH_EXAMPLE FALSE CACHE BOOL "" FORCE)
 SET(MKLDNN_LIBRARY_TYPE STATIC CACHE STRING "" FORCE)
+IF(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
```

Comment:
I guess you specifically treat "GNU" because you want lose less in performance. But this will not save users on older Intel CPUs. From MKL-DNN README:

> By default, Intel MKL-DNN is built specifically for the processor type of the
> compiling machine (for example, `-march=native` in the case of GCC). While this option
> gives better performance, the resulting library can be run only on systems
> that are instruction-set compatible with the compiling machine.
>
> Therefore, **if Intel MKL-DNN is to be shipped to other platforms (for example, built by
> Linux distribution maintainers), consider setting `ARCH_OPT_FLAGS` to `""`**.

Rather, I would rely on `USE_NATIVE_ARCH` to decide whether to empty `ARCH_OPT_FLAGS` and forget about `msse4`.

Reply:
@gujinghui you need GNU and Clang covered, we need the same for OSX.

Reply:
@xuhdev pytorch declaress sse4 as a minimum requirement. We do this elsewhere in our cmake scripts when compiling ATen for example.

Reply:
OK, fair enough. Perhaps we should respect `USE_NATIVE_ARCH` then, otherwise builds using GNU or Clang will never get more optimized than sse4.